app/plugins/request-transformer.md

```diff
@@ -43,12 +43,23 @@ form parameter                            | description
 ---:                                      | ---
 `name`                                    | Name of the plugin to use, in this case: `request-transformer`
 `consumer_id`<br>*optional*               | The CONSUMER ID that this plugin configuration will target. This value can only be used if [authentication has been enabled][faq-authentication] so that the system can identify the user making the request.
-`config.add.headers`<br>*optional*         | Comma separated list of `headername:value` to add to the request headers.
-`config.add.querystring`<br>*optional*     | Comma separated list of `paramname:value` to add to the request querystring.
-`config.add.form`<br>*optional*            | Comma separated list of `paramname:value` to add to the request body in urlencoded format.
 `config.remove.headers`<br>*optional*      | Comma separated list of header names to remove from the request.
 `config.remove.querystring`<br>*optional*  | Comma separated list of parameter names to remove from the request querystring.
 `config.remove.form`<br>*optional*         | Comma separated list of parameter names to remove from the request body.
+`config.replace.headers`<br>*optional*      | Comma separated list of `headername:value` to update the value of specified request header only if it exists.
+`config.replace.querystring`<br>*optional*  | Comma separated list of `paramname:value` to update the value of specified querystring only if it exists.
+`config.replace.form`<br>*optional*         | Comma separated list of `paramname:value` to update the value of existing request parameter only if it exists.
+`config.add.headers`<br>*optional*         | Comma separated list of `headername:value` to add new request header only if specified header doesn't exists.
+`config.add.querystring`<br>*optional*     | Comma separated list of `paramname:value` to add new querystring only if specified querystring doesn't exists.
+`config.add.form`<br>*optional*            | Comma separated list of `paramname:value` to add new parameter to the request body in urlencoded format only if parameter doesn't exists.
+`config.append.headers`<br>*optional*         | Comma separated list of `headername:value` to add to new request header if specified header doesn't exists or to append value to specified request header.
+`config.append.querystring`<br>*optional*     | Comma separated list of `paramname:value` to add new querystring if specified querystring doesn't exists or to append value to specified querystring.
```

Comment:
List of `paramname:value`. If the parameter is not present in the querystring, it will be added with the given key/value. If it is already present, the key/value will be added and the old one preserved (`foo=hello&foo=world`).

This is I think a more descriptive explanation.